lib/logging/index.js

```diff
@@ -0,0 +1,576 @@
+ * @param {module:logging/sink} callback.sink - The created Sink object.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * var gcs = gcloud.storage();
+ *
+ * var config = {
+ *   destination: gcs.bucket('logging-bucket'),
+ *   filter: 'metadata.severity = ALERT'
+ * };
+ *
+ * function callback(err, sink, apiResponse) {
+ *   // `sink` is a Sink object.
+ * }
+ *
+ * logging.createSink('new-sink-name', config, callback);
+ */
+Logging.prototype.createSink = function(name, config, callback) {
+  // jscs:enable maximumLineLength
+  var self = this;
+
+  if (!is.string(name)) {
+    throw new Error('A sink name must be provided.');
+  }
+
+  if (!is.object(config)) {
+    throw new Error('A sink configuration object must be provided.');
+  }
+
+  if (config.destination instanceof Bucket) {
+    this.setAclForBucket_(name, config, callback);
+    return;
+  }
+
+  if (config.destination instanceof Dataset) {
+    this.setAclForDataset_(name, config, callback);
+    return;
+  }
+
+  if (config.destination instanceof Topic) {
+    this.setAclForTopic_(name, config, callback);
+    return;
+  }
+
+  this.request({
+    method: 'POST',
+    uri: '/sinks',
+    json: extend({}, config, {
+      name: name
+    })
+  }, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var sink = self.sink(resp.name);
+    sink.metadata = resp;
+
+    callback(null, sink, resp);
+  });
+};
+
+/**
+ * Create an entry object.
+ *
+ * Note that using this method will not itself make any API requests. You will
+ * use the object returned in other API calls, such as
+ * {module:logging/log#write}.
+ *
+ * @resource [LogEntry JSON representation]{@link https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/LogEntry}
+ *
+ * @param {object=|string=} resource - See a
+ *     [Monitored Resource]{@link https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/MonitoredResource}.
+ * @param {object|string} data - The data to use as the value for this log
+ *     entry.
+ * @return {module:logging/entry}
+ *
+ * @example
+ * var resource = {
+ *   type: 'gce_instance',
+ *   labels: {
+ *     zone: 'global',
+ *     instance_id: 3
+ *   }
+ * };
+ *
+ * var entry = logging.entry(resource, {
+ *   delegate: process.env.USER
+ * });
+ *
+ * entry.toJSON();
+ * // {
+ * //   resource: {
+ * //     type: 'gce_instance',
+ * //     labels: {
+ * //       zone: 'global',
+ * //       instance_id: 3
+ * //     }
+ * //   },
+ * //   jsonPayload: {
+ * //     delgate: process.env.USER
+ * //   }
+ * // }
+ */
+Logging.prototype.entry = function(resource, data) {
+  return new Entry(resource, data);
+};
+
+/**
+ * List the entries in your logs.
+ *
+ * @resource [entries.list API Documentation]{@link https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/entries/list}
+ *
+ * @param {object=} options - Services filtering options.
```

Comment:
The most important option "filter" is missing from the options :)
Also it's rarely a good idea to pass an empty one, as it will return 30 days of logs starting from oldest ones.